### PR TITLE
MRG: Fix multitaper

### LINKED ISSUE #2931: off-by-1 error in notch filter (multitaper)
I've been running into this on occasion when using the multi-taper line noise removal:
ValueError: operands could not be broadcast together with shapes (1,1,245411) (8,245412) 

I'm attaching a log file with the error. Note that this doesn't always happen, but I haven't been able to figure out the specific circumstances that trigger the error.

I've managed to get around this on a temporary basis by commenting out 1142-1143 in filter.py, but that winds up taking quite a bit longer for long pieces of data...

Any ideas?

Log file:
[preprocess_eeg_data.txt](https://github.com/mne-tools/mne-python/files/141563/preprocess_eeg_data.txt)

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #2931.

Upstream bug https://github.com/nipy/nitime/pull/140.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Ready for review/merge from my end.

File: mne/time_frequency/tests/test_multitaper.py [hunk 1]
```diff
@@ -33,6 +33,9 @@ def test_dpss_windows():
     assert_array_almost_equal(dpss, dpss_ni)
     assert_array_almost_equal(eigs, eigs_ni)
 
+    dpss, _ = dpss_windows(245411, 4, 8, False, 1000)
+    assert_equal(dpss.shape[-1], 245411)
```
Comment (agramfort) [inline mne/time_frequency/tests/test_multitaper.py]: is this long to compute?

Comment (Eric89GXL) [inline mne/time_frequency/tests/test_multitaper.py]: ~1 sec. But we could get rid of it if you want. I'm + 0.5 on removing it but I was afraid of no-test complainers :)

Comment (agramfort) [inline mne/time_frequency/tests/test_multitaper.py]: I'd remove it as these lines are covered....

**Comment by Eric89GXL** [comment]
@agramfort done. Merge once CIs are happy?

**Comment by dengemann** [comment]
@Eric89GXL what are the consequences of this bug? Is it in the category {hashtag}recompute_your_results?

**Comment by Eric89GXL** [comment]
The bug made it so that if you used really long signals you'd get a dimension error during multiplication. So if you haven't hit it, then it probably didn't affect you.

I also included another tweak that appeared upstream that determined when to flip the multitaper. I'm not 100% sure of the consequences there, but I'd expect them to be minor.

**Comment by Eric89GXL** [comment]
@dengemann okay to merge for you?

**Comment by dengemann** [comment]
Yes!

**Comment by agramfort** [comment]
+1 for MRG

**Comment by Eric89GXL** [comment]
Thanks for the quick reviews

**Pull request merged**